src/VisualStudio/IntegrationTest/TestUtilities/InProcess/SolutionExplorer_InProc.cs

```diff
@@ -364,6 +365,30 @@ public void CleanUpOpenSolution()
                 }
             });
 
+            if (dte.Debugger.CurrentMode != EnvDTE.dbgDebugMode.dbgDesignMode)
```

Comment:
Consider factoring out into a separate function. It seems like there is a non-zero chance this could result in a hang given the looping. If that does happen and we factor out to a separate function then the cause of the hang will be immediately obvious when analyzing a crash dump. In the current form it will require a bit more digging.

Reply:
Extracted to a helper *and* applied a hang mitigating timeout 😄